great_expectations/core/expectation_suite.py

```diff
@@ -867,6 +906,15 @@ def _build_expectation(
                 f"Could not add expectation; provided configuration is not valid: {e.message}"
             ) from e
 
+    def _add_save_callback_to_expectation(
+        self,
+        expectation: Expectation,
+        expectation_configuration: ExpectationConfiguration,
+    ) -> Expectation:
+        save_callback = partial(self.update_by_identity, id(expectation_configuration))
+        expectation.register_save_callback(callback=save_callback)
+        return expectation
```

Comment:
super nit: the name of this method does not lead me to think I'd get an expectation back. We already have a handle on the expectation; I'd probably just not return anything (unless we are trying for a more fluent-ish flow. Actually looking at the overall flow, I might just include those 2 lines right in `_build_expectation`.

Reply:
this is actually called a couple other places, since we need to update the ref on CRUD methods. will leave as is for now, but there might be a clearer pattern in the future